cmd/kube-proxy/app/server.go

```diff
@@ -237,6 +237,9 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
 		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
 		iptables.CleanupLeftovers(iptInterface)
 	}
+	if err != nil && proxier == nil {
```

Comment:
glog.Fatalf terminates the program.  Your errInf assignments can never
trigger.  I get the sense that you're solving a hypothetical problem. Do
you have a real issue that is triggering?

On Jul 18, 2016 6:26 PM, "KeZhang" <notifications@github.com> wrote:

> In cmd/kube-proxy/app/server.go
> <https://github.com/kubernetes/kubernetes/pull/29089#discussion_r71259721>
> :
>
> > @@ -237,6 +237,9 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
> >  		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
> >  		iptables.CleanupLeftovers(iptInterface)
> >  	}
> > +	if err != nil && proxier == nil {
>
> Sorry @thockin <https://github.com/thockin> , i had a mistake on the err
> variable, now i have changed the code, please review again, thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29089/files/f411a9ca89413720fb245d16d0d1c41346ea5041#r71259721>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVHLpDmbGFTOu5iiVLgB6FY3SEAmhks5qXCewgaJpZM4JOd8K>
> .
>